libtiledbsoma/test/test_indexer.cc

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file   test_indexer.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file manages unit tests for column buffers
+ */
+
+#include <reindexer/reindexer.h>
+#include <catch2/catch_test_macros.hpp>
+#include <cstdint>
+#include <string>
+#include <tiledb/tiledb>
+#include <unordered_map>
+#include <vector>
+#include "external/khash/khash.h"
+
+namespace {
+
+/**
+ * @brief Create multiple 64 bit vector and try the reindexer compared to khash
+ * as baseline.
+ *
+ */
+
+KHASH_MAP_INIT_INT64(m64, int64_t)
+
+bool run_test(std::vector<int64_t> keys, std::vector<int64_t> lookups) {
+    std::vector<int64_t> indexer_results;
+    indexer_results.resize(lookups.size());
+
+    tiledbsoma::IntIndexer indexer;
+    indexer.map_locations(keys);
+    auto* hash = kh_init(m64);
+    int ret;
+    khint64_t k;
+
+    for (size_t i = 0; i < keys.size(); i++) {
+        k = kh_put(m64, hash, keys[i], &ret);
+        assert(k != kh_end(hash));
+        kh_val(hash, k) = i;
+    }
```

Comment:
```suggestion
    for (auto key& : keys) {
        k = kh_put(m64, hash, key, &ret);
        assert(k != kh_end(hash));
        kh_val(hash, k) = i;
    }
```

Reply:
Nevermind about this. I missed that this was a pointer.

Reply:
Actually this one is not a pointer but i is used in the loop so writing this way is more intuitive.